### PR TITLE
[feature-wip](MTMV)optimize lock

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #xxx

optimize lock

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVJobManager.java [hunk 1]
```diff
@@ -470,56 +317,75 @@ public void replayDropJobs(List<Long> jobIds) {
...
     public void replayCreateJobTask(MTMVTask task) {
         taskManager.replayCreateJobTask(task);
     }
 
     public void replayDropJobTasks(List<String> taskIds) {
-        taskManager.dropTasks(taskIds, true);
+        taskManager.dropHistoryTasks(taskIds, true);
     }
 
     public void removeExpiredJobs() {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVJobManager.java:334]: change to private.
And check other method, if it only be used inside this class, change to private

File: fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVJobManager.java [hunk 2]
```diff
@@ -470,56 +317,75 @@ public void replayDropJobs(List<Long> jobIds) {
...
-
-                }
-                long expireTime = job.getExpireTime();
-                if (expireTime > 0 && currentTimeSeconds > expireTime) {
-                    jobIdsToDelete.add(job.getId());
-                }
+        List<MTMVJob> jobs = getAllJobsWithLock();
+        for (MTMVJob job : jobs) {
+            // active job should not clean
+            if (job.getState() == MTMVUtils.JobState.ACTIVE) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVJobManager.java:340]: I think only `COMPLETE` job can be removed.

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by xxiao2018 — APPROVED** [review]
LGTM

**Pull request merged**